### PR TITLE
Add spring-boot sample

### PR DESCRIPTION
**Pull request by ramsrib**
### Context
Added spring-boot sample which is missing in the present kotlin-dsl samples.

### Contributor Checklist
- [x] Base the PR against the `develop` branch
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Provide integration tests to verify changes from a user perspective
- [x] Provide unit tests to verify logic
- [x] Ensure that tests pass locally: `./gradlew check --parallel`

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
I think you need to rebase your changes on `develop`.

File: samples/hello-spring-boot/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "com.spring.boot"
+version = "1.0.0-SNAPSHOT"
+
+plugins {
+    val kotlinVersion = "1.2.21"
+    id("org.springframework.boot") version "2.0.0.RELEASE"
+    id("org.jetbrains.kotlin.jvm") version kotlinVersion
+    id("org.jetbrains.kotlin.plugin.spring") version kotlinVersion
```
Comment (StefMa) [inline samples/hello-spring-boot/build.gradle.kts]: Could be replaced with `kotlin("plugin.spring") version kotlinVersion`
(same for `jvm` of course)

File: samples/hello-spring-boot/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+tasks.withType<KotlinCompile> {
+    kotlinOptions {
+        jvmTarget = "1.8"
+        freeCompilerArgs = listOf("-Xjsr305=strict")
+    }
+}
+
+val test by tasks.getting(Test::class) {
```
Comment (StefMa) [inline samples/hello-spring-boot/build.gradle.kts]: Could be also `(tasks.findByName("test") as Test).useJUnitPlatform()`.
From my point of view it looks better... But don't know 🤷‍♂️

File: samples/hello-spring-boot/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile("org.springframework.boot:spring-boot-starter-web")
+    compile("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    compile("org.jetbrains.kotlin:kotlin-reflect")
```
Comment (StefMa) [inline samples/hello-spring-boot/build.gradle.kts]: You can here use `kotlin()` as well:
```kotlin
compile(kotlin("stdlib-jdk8"))
compile(kotlin("reflect"))
```

**Comment by ramsrib** [comment]
I created this patch from master, so i'll drop this PR and create one from develop branch to avoid conflicts. 

Also i'll include the review comments in that new PR (https://github.com/gradle/kotlin-dsl/pull/746).

**Pull request closed without merging**